### ISSUE TITLE
fluent-ffmpeg-filters

### ISSUE THREAD

**Issue by itoche** [fluent-ffmpeg/node-fluent-ffmpeg#812]
Hi,

FYI, some time ago I made the following project to extend node-fluent-ffmpeg to allow using a fluent API for configuring the filters as well.

https://github.com/itoche/fluent-ffmpeg-filters

Let me know what you think of it.

**Comment by njoyard** [comment]
Hey @itoche,

This is really nice. Why didn't you just propose a PR to integrate this into fluent-ffmpeg ? ;)

My only concern at this point is the fact that the generated code is tied to the documentation, and thus to the latest ffmpeg version available at the time the generation was run. I feel like we could do this more dynamically using the outputs from `ffmpeg -filters` and `ffmpeg -h filter=<filtername>`:

```sh
$ ffmpeg -filters 2>/dev/null
Filters:
  T.. = Timeline support
  .S. = Slice threading
  ..C = Command support
  A = Audio input/output
  V = Video input/output
  N = Dynamic number and/or type of input/output
  | = Source or sink filter
 ... abench            A->A       Benchmark part of a filtergraph.
 ... acompressor       A->A       Audio compressor.
 ... acopy             A->A       Copy the input audio unchanged to the output.
 ... acrossfade        AA->A      Cross fade two input audio streams.
 ... acrusher          A->A       Reduce audio bit resolution.
 T.. adelay            A->A       Delay one or more audio channels.
 ... aecho             A->A       Add echoing to the audio.
 ... aemphasis         A->A       Audio emphasis.
 ... aeval             A->A       Filter audio signal according to a specified expression.
 T.. afade             A->A       Fade in/out input audio.
 ... afftfilt          A->A       Apply arbitrary expressions to samples in frequency domain.
 .S. afir              AA->A      Apply Finite Impulse Response filter with supplied coefficients in 2nd stream.
 ... aformat           A->A       Convert the input audio to one of the specified formats.
 ... agate             A->A       Audio gate.
 ... ainterleave       N->A       Temporally interleave audio inputs.
 ... alimiter          A->A       Audio lookahead limiter.
[snip]
```

```sh
$ ffmpeg -h fiter=vstack 2>/dev/null
Filter vstack
  Stack video inputs vertically.
    Inputs:
        dynamic (depending on the options)
    Outputs:
       #0: default (video)
vstack AVOptions:
  inputs            <int>        ..FV.... set number of inputs (from 2 to INT_MAX) (default 2)
  shortest          <boolean>    ..FV.... force termination when the shortest input terminates (default false)
```

I also feel the generator code is undercommented and a bit hard to follow at first glance.

But again, this is a really interesting approach. Nice work :)

**Comment by itoche** [comment]
Thanks for the feedback.

I'm also not very proud of the generator code :) 

Your suggestion is much smarter. I think  I could work on it if you can wait a bit (this is a side project of a side project for me :))
